libsolidity/parsing/Token.cpp

```diff
@@ -151,20 +153,19 @@ tuple<Token::Value, unsigned int, unsigned int> Token::fromIdentifierOrKeyword(s
 				positionM < positionX &&
 				positionX < _literal.end() &&
 				*positionX == 'x' &&
-				all_of(positionX + 1, _literal.end(), ::isdigit)
+				all_of(++positionX, _literal.end(), ::isdigit)
```

Comment:
I made some changes here that are reverted now. Are you sure your changes are correct?